### ISSUE TITLE
Allow us to get the elementRef

### ISSUE THREAD

**Issue by Venrir** [ncstate-sat/popover#144]
Would be a nice feature if we could access the container to allow us to use their properties, (height, width, etc).

because i want to overflow the popover with the anchorWidth + 50px

which looks pretty nice. (Firebase table edit box)

![img](https://i.mavis.moe/f/9M63xf2bg6/chrome-2018-12-11-14-02-20.png)

**Comment by willshowell** [comment]
To be clear, you want the `elementRef` of the `SatPopoverAnchor` directive, right? If so, this sounds like a fine thing to add.

**Comment by willshowell** [comment]
This should be pretty easy to do already unless you have a use case/pattern I'm not considering:

https://stackblitz.com/edit/sat-popover-issues-qkjyed?file=app%2Fapp.component.html

I'm going to add a `SatPopoverAnchor.getElement` method just to make it a little easier to get the `elementRef` when the anchor is a component.

**Comment by Venrir** [comment]
would be nice if there is an option to match the anchors width and height.
also according to your demo the popover does not align it self properly after setting their width/offset property.

**Comment by willshowell** [comment]
> would be nice if there is an option to match the anchors width and height

The popover itself doesn't have a height or width (it simply wraps whatever content the user projects inside) so I'd rather leave that up to users to implement. You could write a directive that takes in the anchor and does it for you.

```
<div #anchor [satAnchorFor]="popover">Anchor</div>

<sat-popover #popover>
  <app-my-content [appCoverElement]="anchor" appCoverElementXOffset="50"> ... </app-my-content>
</sat-popover>
```

> the popover does not align it self properly after setting their width/offset property

Yeah, this is a bummer. I will look into adding a `popover.realign()` method.

**Comment by willshowell** [comment]
These features have been added in 62d5d4a and 4e248c1. They'll be available in v3.1 shortly
